components/core/src/Utils.cpp

```diff
@@ -213,3 +219,124 @@ ErrorCode read_list_of_paths (const string& list_path, vector<string>& paths) {
 
     return ErrorCode_Success;
 }
+
+void load_lexer_from_file (std::string schema_file_path,
+                           bool reverse,
+                           log_surgeon::lexers::ByteLexer& lexer) {
+    FileReader schema_reader;
+    schema_reader.try_open(schema_file_path);
+    /// TODO: this wrapper is repeated a lot
+    log_surgeon::Reader reader_wrapper{[&](char* buf, size_t count, size_t& read_to) -> log_surgeon::ErrorCode {
+        schema_reader.read(buf, count, read_to);
+        if (read_to == 0) {
+            return log_surgeon::ErrorCode::EndOfFile;
+        }
+        return log_surgeon::ErrorCode::Success;
+    }};
```

Comment:
Maybe lets add a helper function to FileReader, that we can pass into `schema_parser.generate_schema_ast`?
(Another option is to wrap `generate_schema_ast` and take a `SchemaParser` as a parameter.)

Reply:
commit a0088824a457364bcb92d12d298a0db7fd3d1dcf